extensions/ql-vscode/src/data-extensions-editor/model-details/model-details-data-provider.ts

```diff
@@ -61,12 +61,45 @@ export class ModelDetailsDataProvider
       return [];
     }
   }
+
+  getParent(
```

Comment:
I investigated a few different ways we could implement `getParent` without having to search through all items, but none seemed fantastic at the time. I might have another go at it though.

Adding a literal link to the `ExternalApiMethod` object to the `Usage` object seemed a bit ugly and fiddly. Another option would be the add the `ExternalApiMethod.signature` field to the `Usage` object, so then it's only adding a string field and we can easily use that key to look up the `ExternalApiMethod`.

Either way, we could have new types just for the data in this class, so we don't have to add new fields to the types used by the rest of the application.